test/cpp_extensions/msnpu_extension.cpp

```diff
@@ -55,6 +55,15 @@ Tensor ones_like_override(const Tensor & self, const TensorOptions & options) {
   return get_dtype_tensor(options.dtype());
 }
 
+Tensor clamp_override(const Tensor & self, c10::optional<Scalar> min, c10::optional<Scalar> max) {
```

Comment:
we do check the function signature though, right?  i.e. if you tried to register `clamp_override(const Tensor & self, Scalar min, Scalar max)` it would complain, no?

Reply:
It silently "works."
E.g. there's an msnpu test which registered a function that takes ScalarType. If the parameter is changed in nativefunctions to to be optional<ScalarType>=nullopt, the actual value for dtype ends up being out of range of the enumerated values. By adding the '?', the existing registration fails, making it more obvious that the function needs to be changed.